nms/packages/magmalte/app/views/subscriber/SubscriberTable.js

```diff
@@ -161,8 +169,174 @@ async function exportSubscribers(props: ExportProps) {
     });
   }
 }
+const StyledMenu = withStyles({
+  paper: {
+    border: '1px solid #d3d4d5',
+  },
+})(props => (
+  <Menu
+    elevation={0}
+    getContentAnchorEl={null}
+    anchorOrigin={{
+      vertical: 'bottom',
+      horizontal: 'center',
+    }}
+    transformOrigin={{
+      vertical: 'top',
+      horizontal: 'center',
+    }}
+    {...props}
+  />
+));
+function SubscriberActionsMenu(props: {onClose: () => void}) {
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const [open, setOpen] = React.useState(false);
+  const [error, setError] = React.useState('');
+  const enqueueSnackbar = useEnqueueSnackbar();
+  const ctx = useContext(SubscriberContext);
+  const successCountRef = useRef(0);
+  const handleClick = event => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
 
-function SubscribersTable(props: WithAlert & {refresh: boolean}) {
+  const saveSubscribers = async (
+    subscribers: Array<SubscriberInfo>,
+    edit: boolean,
+  ) => {
+    let addedSubscribers = [];
+    let subscriberErrors = '';
+    for (const [i, subscriber] of subscribers.entries()) {
+      try {
+        const authKey =
+          subscriber.authKey && isValidHex(subscriber.authKey)
+            ? hexToBase64(subscriber.authKey)
+            : '';
+
+        const authOpc =
+          subscriber.authOpc !== undefined && isValidHex(subscriber.authOpc)
+            ? hexToBase64(subscriber.authOpc)
+            : '';
+        const newSubscriber = {
+          active_apns: subscriber.apns,
+          active_policies: subscriber.policies,
+          id: subscriber.imsi,
+          name: subscriber.name,
+          lte: {
+            auth_algo: 'MILENAGE',
+            auth_key: authKey,
+            auth_opc: authOpc,
+            state: subscriber.state,
+            sub_profile: subscriber.dataPlan,
+          },
+        };
+        if (edit) {
+          ctx.setState?.(subscriber.imsi, newSubscriber);
+        } else {
+          addedSubscribers.push(newSubscriber);
+          // bulk add chunked subscribers
+          if (
+            addedSubscribers.length == SUBSCRIBERS_CHUNK_SIZE ||
+            i == subscribers.length - 1
+          ) {
+            const success = await bulkAdd(addedSubscribers, subscriberErrors);
+            if (success) {
+              successCountRef.current =
+                successCountRef.current + addedSubscribers.length;
+              addedSubscribers = [];
+            } else {
+              enqueueSnackbar('Saving subscribers to the api failed1: ', {
+                variant: 'error',
+              });
+              return;
+            }
+          }
+        }
+      } catch (e) {
+        const errMsg = e.response?.data?.message ?? e.message ?? e;
+        subscriberErrors +=
+          'error saving ' + subscriber.imsi + ' : ' + errMsg + '\n';
```

Comment:
Capitalization - these errors are displayed to the user
```suggestion
          'Error saving ' + subscriber.imsi + ' : ' + errMsg + '\n';
```